cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2276,7 +2326,12 @@ private List<Field<?>> getProgramRunInvertedTimeKey(String recordType, ProgramRu
                                                       long startTs, boolean includeVersion) {
```

Comment:
should we have a ProgramRunReference class as well?

Reply:
adding ProgramRunReference class would need some refactoring since in many places where runId is queried, and we should ignore the version. This should probably be in a separate PR, added a TODO comment[JIRA CDAP-20031](https://cdap.atlassian.net/browse/CDAP-20031) to follow up on this.